internal/cmd/controller/reconciler/cluster_controller.go

```diff
@@ -144,20 +151,25 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	if err != nil {
 		return ctrl.Result{}, r.updateErrorStatus(ctx, req.NamespacedName, cluster.Status, err)
 	}
-
-	deleted := map[types.UID]bool{}
-	for _, bundle := range cleanup {
-		for _, bd := range bundleDeployments.Items {
-			if bd.Labels[fleet.BundleLabel] == bundle.Name && bd.Labels[fleet.BundleNamespaceLabel] == bundle.Namespace {
-				logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
-				err := r.Delete(ctx, &bd)
-				if err != nil {
-					logger.V(1).Info("deleting bundleDeployment returned an error", "error", err)
-				}
-				deleted[bd.GetUID()] = true
+	toDeleteBundles := indexByNamespacedName(cleanup)
+
+	// Delete BundleDeployments for to Bundles being removed while getting a filtered items list
+	bundleDeployments.Items = slices.DeleteFunc(bundleDeployments.Items, func(bd fleet.BundleDeployment) bool {
+		bundleNamespace := bd.Labels[fleet.BundleNamespaceLabel]
+		bundleName := bd.Labels[fleet.BundleLabel]
+		if _, ok := toDeleteBundles[types.NamespacedName{Namespace: bundleNamespace, Name: bundleName}]; ok {
+			logger.V(1).Info("cleaning up bundleDeployment not matching the cluster", "bundledeployment", bd)
+			if err := r.Delete(ctx, &bd); err != nil {
+				logger.V(1).Info("deleting bundleDeployment returned an error", "error", err)
 			}
+			return true
 		}
-	}
+		// Omit fleet-agent bundle deployments from resource counts
+		if strings.HasPrefix(bd.Name, "fleet-agent") {
+			return true
+		}
```

Comment:
This has an effect on the calculations performed later, since `fleet-agent` bundles (and its k8s resources) will be omitted from any calculations and display. This is also reflected in the e2e that started failing:
```
Checks status updates happen for a simple deployment when deployment is successful [It] correctly sets the status values for Clusters
...
  Expected
      <string>: '1/1'
  to equal
      <string>: '2/2'
```
However, the actual test expectation would not be obvious if you don't know that fleet-agents are deployed using bundles (e.g. why it expects 2 ready bundles if I only deployed 1?).
I thought users may have the same confusion, and I added this for that reason, but happy to revert if if you think otherwise.

Reply:
I think this approach of making Fleet agent deployments transparent to users make sense, but am curious to know what others think. Either way, we should document this.

Reply:
+1
For completeness, this only affects the Cluster resource counts. They will still appear under Bundles in the Advanced view

Reply:
Consensus was to keep including fleet-agent for consistency.